### PR TITLE
Get sdl2 building on Fedora Linux 28

### PR DESCRIPTION
**Pull request by madebr**
- Get it building on Fedora Linux 28
- The separate steps work (`conan source`, `conan install`, `conan build`)
    - `conan build` modifies the sources, so `conan source` should be rerun after every `conan build`

### CODE REVIEW

**Comment by SSE4** [comment]
so, the only package differs is pkg-config?

**Comment by madebr** [comment]
Yes. I haven't enabled any other options.

File: conanfile.py [hunk 1]
```diff
@@ -264,7 +263,7 @@ def build_cmake(self):
         elif self.settings.os == "Windows":
             cmake.definitions["DIRECTX"] = self.options.directx
 
-        cmake.configure(build_dir='build')
+        cmake.configure(build_dir=self.install_folder)
```
Comment (Croydon) [inline conanfile.py:266]: self.install_folder doesn't seem to be defined (linter gave me the hint on export)

**Review by Croydon — DISMISSED** [review]
I'm testing on Fedora 28 as well.

`sudo yum check-update` fails for me when I do have - unrelated - system packages which are not up to date. This looks like a massive system invasion, which is completely unneccessary. From my understanding this might be rather a Conan bug? 

After a system wide update of my packages I'm getting to:

>CMake Error at source_subfolder/CMakeLists.txt:2 (message):
  Prevented in-tree built.  Please create a build directory outside of the
  SDL source code and call cmake from there

which is likely the cause of undefined self.install_folder

**Comment by SSE4** [comment]
@Croydon yep, it's conan bug, see https://github.com/conan-io/conan/issues/3180

File: conanfile.py (hunk 1 above)
Comment (madebr) [reply conanfile.py:266]: Idk, but while debugging the conan SDL2 build process, self.source_folder, self.build_folder, self.install_folder and self.package_folder are available for SDL2Conan.
Those seem to be the only member variables of SDL2Conan with file paths.

Comment (madebr) [reply conanfile.py:266]: https://docs.conan.io/en/latest/reference/conanfile/attributes.html#folders

**Comment by SSE4** [comment]
@Croydon @madebr 
I've pushed changes to system_requirements only for now: 
https://github.com/bincrafters/conan-sdl2/commit/a3a79ffd76361ad93399e3384abcd0e6cfee11e2
tested on **bincrafters/docker-fedora-gcc7** docker image.
changes to CMake introduced breakage of appveyor/travis, so for now I avoid them.

**Comment by madebr** [comment]
The failure is because the bincrafters script is doing `conan install` and `conan build` in the source directory.
The current build script expects a `build` directory to exist inside the source directory.
My modification does not. It expects the install folder to be a good path. (see the `-if` and `-bf` switches)

**Comment by SSE4** [comment]
@madebr I'll need to think on better solution for the CMake, which will satisfy your use-case and will not break appveyor/travis at the same time.

**Comment by madebr** [comment]
I'm talking about supporting this flow: https://docs.conan.io/en/latest/developing_packages/package_dev_flow.html
imho, every conan package should support this

**Comment by SSE4** [comment]
@madebr I've added changes to better support conan package development flow:
https://github.com/bincrafters/conan-sdl2/commit/15fc9eb80ddf38d6e92bc270b2fba524031ceb22
commands I've used:
```
rm -rf $HOME/sdl-source $HOME/sdl-build $HOME/sdl-package
conan source . --source-folder=$HOME/sdl-source
conan install . --install-folder=$HOME/sdl-build
conan build . --source-folder=$HOME/sdl-source --build-folder=$HOME/sdl-build
conan package . --source-folder=$HOME/sdl-source --build-folder=$HOME/sdl-build --package-folder=$HOME/sdl-package
conan export-pkg . bincrafters/testing --package-folder=$HOME/sdl-package -f
conan test test_package sdl2/2.0.8@bincrafters/testing
```

NOTE that conan (1.5.2) produces linter warning for **self.install_folder**, despite attribute is available and documented:
```
Linter warnings
    WARN: Linter. Line 244: Instance of 'SDL2Conan' has no 'install_folder' member
```

**Comment by SSE4** [comment]
@Croydon @madebr  I am closing this one out, as changes for both Fedora support and conan development workflow support were applied. if you still experience any problems, feel free to open issue.

**Pull request closed without merging**